steel/src/contract.rs

```diff
@@ -286,13 +289,15 @@ where
     ///
     /// [EvmEnv]: crate::EvmEnv
     pub fn call(self) -> C::Return {
```

Comment:
Side note that it'd be good to use an error type that (when `std` is enabled) implements `std::error::Error`. We can address that later though (or not at all. It's not critical)
```suggestion
    pub fn call(self) -> Result<C::Return, String> {
```

Reply:
100% agree, after all it should be a `thiserror` or something. However, it was not that trivial to do, so I decided to leave it as a string for now.